src/v2i-hub/ImmediateForwardPlugin/test/TestIMFNTCIP1218Worker.cpp

```diff
@@ -0,0 +1,169 @@
+    // Call the function
+    snmp_response_obj obj;
+    obj.type = snmp_response_obj::response_type::INTEGER;
+    obj.val_int = 2;
+    EXPECT_CALL( *mockClient, process_snmp_request(rsu::mib::ntcip1218::rsuModeOid, request_type::SET , obj) ).Times(1).WillRepeatedly(testing::DoAll(
+        testing::SetArgReferee<2>(obj),
+        Return(true)));
+    
+    setRSUMode(mockClient.get(), 2);
+}
+
+TEST(TestIMFNTCIP1218Worker, testSetRSUModeFail) {
+    // Test the setRSUMode function
+    // Create a mock SNMP client
+    std::unique_ptr mockClient = std::make_unique<mock_snmp_client>("", 0, "", "", "", "");
+    // Call the function
+    snmp_response_obj obj;
+    obj.type = snmp_response_obj::response_type::INTEGER;
+    obj.val_int = 2;
+    EXPECT_CALL( *mockClient, process_snmp_request(rsu::mib::ntcip1218::rsuModeOid, request_type::SET , obj) ).Times(1).WillRepeatedly(testing::DoAll(
+        testing::SetArgReferee<2>(obj),
+        Return(false)));
+    
+    EXPECT_THROW(setRSUMode(mockClient.get(), 2), tmx::TmxException);
+}
+
+TEST(TestIMFNTCIP1218Worker, testInitializeImmediateForwardTable) {
+    // Test the initializeImmediateForwardTable function
+    // Create a mock SNMP client
+    std::unique_ptr mockClient = std::make_unique<mock_snmp_client>("", 0, "", "", "", "");
+    // Call the function
+    std::vector<Message> messages;
+    Message message;
+    message.tmxType = "SPAT-P";
+    message.sendType = "SPAT";
+    message.psid = "0x8002";
+    message.channel = 183;
+    messages.push_back(message);
+
+    std::vector<snmpRequest> requests ;
+    EXPECT_CALL( *mockClient, process_snmp_set_requests(_) ).WillOnce(testing::DoAll(::testing::SaveArg<0>(&requests), Return(true)));
+    // Expect these snmpRequests to be called
+    // snmpRequest psid{
+    //     rsu::mib::ntcip1218::rsuIFMPsidOid + "." + std::to_string(1),
+    //     'x',
+    //     "8002"
+    // };
+    // snmpRequest channel{
+    //     rsu::mib::ntcip1218::rsuIFMTxChannelOid + "." + std::to_string(1),
+    //     'i',
+    //     "183"
+    // };
+    // snmpRequest enable{
+    //     rsu::mib::ntcip1218::rsuIFMEnableOid + "." + std::to_string(1),
+    //     'i',
+    //     "1"
+    // };
+    // snmpRequest payload{
+    //     rsu::mib::ntcip1218::rsuIFMPayloadOid + "." + std::to_string(1),
+    //     'x',
+    //     "FE"
+    // };
+    // snmpRequest creatRow{
+    //     rsu::mib::ntcip1218::rsuIFMStatusOid + "." + std::to_string(1),
+    //     'i',
+    //     "4"
+    // };
+    // snmpRequest priority{
+    //     rsu::mib::ntcip1218::rsuIFMPriorityOid + "." + std::to_string(1),
+    //     'i',
+    //     "6"
+    // };
+
+    // snmpRequest options{
+    //     rsu::mib::ntcip1218::rsuIFMOptionsOid + "." + std::to_string(1),
+    //     'x',
+    //     "01"
+    // };
+    initializeImmediateForwardTable(mockClient.get(), messages);
+    EXPECT_EQ(requests.size(), 7);
```

Comment:
How is this requests here populated with those  individual request commented out?

Reply:
Shouldn't we assign the returned value from initializeImmediateForwardTable() to requests? also is it "snmpRequest" or "snmp_request"?